Adds stats to creation/destruction of cleanable decals and trash on station

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Keeps track of how many messes are created and removed on the station. I'm adding this because I don't think janitor roles can keep on top of the mess on the station with the way their equipment is currently balanced and I'd like to see what the difference is between messes created and messes destroyed.

I've never added stats before, so I might've done something wrong (I don't even know if I'm allowed to add stats given it seems like more of a database/administration thing). If there's a better way of doing this, let me know.

Let me know if there's a common item of trash I've missed.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

It's useful for seeing how wide the margin is between the creation and removal of mess on the station.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Adds stat tracking for the creation/destruction of trash and cleanable decals (e.g. blood)
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

SpaceManiac commented:
You should check [Setting your commit email address](https://help.github.com/en/articles/setting-your-commit-email-address) for instructions for associating your commits with your GitHub account without publicizing an email address.

JJRcop commented on code/game/objects/items/trash.dm:
```diff
@@ -7,6 +7,16 @@
 	w_class = WEIGHT_CLASS_TINY
 	resistance_flags = FLAMMABLE
 
+/obj/item/trash/Initialize()
```
You may want to consider using the mapload arg to skip trash that is placed by the map, unless that is ok with you.

ExcessiveUseOfCobblestone replied:
u should put in mapload regardless 8)

cacogen replied:
The janitor has to clean up whatever starts on the level. The exception would be maint, which is probably where most roundstart trash begins. I wonder if there's a maint areas define.

cacogen replied:
What will adding mapload do? @ExcessiveUseOfCobblestone

JJRcop replied:
I'm not cobby but it defines the arg for consistency even if you don't use it. Can't see another reason

ExcessiveUseOfCobblestone replied:
yes

cacogen committed:
Adds stat tracking to mess on station

cacogen committed:
Checks for maint to make sure stats aren't counting maint trash which is irrelevant to the janitor

Jordie0608 committed:
adds mapload arg to init

Jordie0608 commented on code/game/objects/items/trash.dm:
```diff
@@ -7,6 +7,18 @@
+
+/obj/item/trash/Destroy()
+	var/area/A = get_area(src)
+	if(is_station_level(initial(z)) && !istype(A, /area/maintenance))
```
Actually I think this won't work as i.e. trash from eating food gets created in a mob's hands, where its z would be 0.

ExcessiveUseOfCobblestone replied:
get the turf then check the turf's z?

ExcessiveUseOfCobblestone replied:
you can prob just check `blob_allowed` on the area and if there's any non-station areas they should have it set to off.

cacogen replied:
@jordie0608 Is there a more sensible way of doing this instead of adding code to the initalise and destroy of every trash item? For example, having a list of trash items that can be updated on the fly and just adding code in /obj/item/Initialize() and /obj/item/Destroy() to check if the item created is one of them?

ExcessiveUseOfCobblestone replied:
why would you not want certain trash items not counted in the "trashed made/clean" stat?

cacogen replied:
Because if I'm adding this stat to see if the janitor can keep on top of mess it doesn't make sense to me to count trash in maintenance.

ExcessiveUseOfCobblestone replied:
I'm not understanding why you wouldn't count trash in maintenance. Janitor should be keeping that clean too since people are expected to traverse it.

cacogen replied:
That is definitely never happening. But I'll remove the check for maint.

ExcessiveUseOfCobblestone commented:
Is the recycler in a room that's considered maint?

cacogen committed:
asdasd

SpaceManiac commented:
@cacogen master is failing Travis, please fix

```
[21:15:18] Runtime in glass.dm,274: Cannot read null.z
  proc name: Initialize (/obj/item/shard/Initialize)
  src: the shard (/obj/item/shard)
  src.loc: null
  call stack:
  the shard (/obj/item/shard): Initialize(1)
  Atoms (/datum/controller/subsystem/atoms): InitAtom(the shard (/obj/item/shard), /list (/list))
  the shard (/obj/item/shard): New(1)
  the glass table (/obj/structure/table/glass): Initialize(1)
  Atoms (/datum/controller/subsystem/atoms): InitAtom(the glass table (/obj/structure/table/glass), /list (/list))
  Atoms (/datum/controller/subsystem/atoms): InitializeAtoms(null)
  Atoms (/datum/controller/subsystem/atoms): Initialize(765146)
  Master (/datum/controller/master): Initialize(10, 0, 1)
```
